File: Angular2Testing/Data/InitialAmount.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Angular2Testing.Data
{
    [Table("ItemDetail.InitialAmount")]
    public partial class InitialAmount
    {
        [Key]
        public int PkID { get; set; }

        [Required]
        [StringLength(75)]
        public string UserName { get; set; }

        public double? Amount { get; set; }

        [Column(TypeName = "date")]
        public DateTime? BeginDate { get; set; }
    }
}

File: Angular2Testing/Areas/ItemDetail/tsScripts/InitialAmount/Models/InitialAmount.ts
/// ===========================================================================================
/// Initial Amount Model
/// ===========================================================================================
export interface IInitialAmount {
    PkID: number;
    UserName: string;
    Amount: number;
    BeginDate: Date;
}

export class InitialAmount implements IInitialAmount {
    constructor(
        public PkID: number,
        public UserName: string,
        public Amount: number,
        public BeginDate: Date
    ) {
    }
}

